src/qat/purr/integrations/qasm.py

```diff
@@ -983,24 +985,30 @@ def visit_QuantumGate(self, node: ast.QuantumGate, context: QasmContext):
             case "u" | "U":
                 theta, phi, _lambda = arguments
                 return self.add_unitary(theta, phi, _lambda, target_qubits, self.builder)
-            case "cu" | "CU":
-                raise NotImplementedError("Controlled Unitary gate is not yet supported.")
             case "cx" | "CX":
                 return self.add_cnot(*target_qubits, self.builder)
             case "ecr" | "ECR":
                 return self.add_ecr(target_qubits, self.builder)
 
+        if len(node.modifiers) > 0:
+            raise NotImplementedError("Gate modifiers are not yet supported.")
+
         gate_context = QasmContext(
             Registers(),
             context.gates,
-            dict(context.variables),
+            dict(),
```

Comment:
intuitively speaking, this is a fresh clean context, not sure if this is intentional, and whether a gate would not need access to prior defined variables

Reply:
Because of the way gate definitions look, they all use the same variables for the input qubits which causes issues when the definition needs to "redefine" what qubit is referred to.